### PR TITLE
Prevent externalized adapters from breaking build

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Closes https://github.com/withastro/astro/issues/11700
- A different approach than https://github.com/withastro/astro/pull/11702, makes a virtual module that can be imported even if the adapter is external.

## Testing

Preview release

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
!preview-si-crypto-reg

**Comment by matthewp** [comment]
!preview si-crypto-reg

File: packages/astro/src/core/build/plugins/plugin-ssr.ts [hunk 1]
```diff
@@ -49,6 +52,9 @@ function vitePluginSSR(
 			if (id === SSR_VIRTUAL_MODULE_ID) {
 				return RESOLVED_SSR_VIRTUAL_MODULE_ID;
 			}
+			if (id === ADAPTER_VIRTUAL_MODULE_ID) {
+				return RESOLVED_ADAPTER_VIRTUAL_MODULE_ID;
```
Comment (ascorbic) [inline packages/astro/src/core/build/plugins/plugin-ssr.ts:56]: Would it work to just return `adapter.serverEntrypoint` here?

**Comment by matthewp** [comment]
Confirmed this fixes the reproduction.

File: packages/astro/src/core/build/plugins/plugin-ssr.ts (hunk 1 above)
Comment (matthewp) [reply packages/astro/src/core/build/plugins/plugin-ssr.ts]: Oh, interesting idea, let me try.

Comment (matthewp) [reply packages/astro/src/core/build/plugins/plugin-ssr.ts]: Doesn't work, rollup complains about it being external again.

**Review by Princesseuh — APPROVED** [review]
don't know everything here but looks fine to me!

**Pull request merged**